buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/ElasticsearchNode.java

```diff
@@ -57,24 +53,22 @@
 import java.util.stream.Stream;
 
 import static java.util.Objects.requireNonNull;
-import static java.util.concurrent.TimeUnit.MILLISECONDS;
-import static java.util.concurrent.TimeUnit.SECONDS;
 
-public class ElasticsearchNode {
+public class ElasticsearchNode implements TestClusterConfiguration {
 
     private final Logger logger = Logging.getLogger(ElasticsearchNode.class);
     private final String name;
-    private final GradleServicesAdapter services;
+    final GradleServicesAdapter services;
```

Comment:
There's no need for these to be package scoped. Reading these values from the first node in order to instantiate subsequent nodes is a bit of an odd pattern. If the cluster needs to retain this info to create subsequent clusters then it should hold onto that state itself rather than rip it off another node.

Reply:
We could improve how Node and Cluster are encapsulated and how they collaborate, can't argue with that, but give how intimate these are and everything is still encapsulated at the package level + this is a small package, does all the extra boilerplate buy us much ? Especially since we might change it over and over as we move projects to use it. I'm more focused on the package level encapsulation. The internals will change allot. At some point we will probably have to separate the cluster configuration from the implementation of starting up a cluster, e.x. to support docker nodes in test-clusters,  or external (cloud) nodes stated by some other means but still verifying that these have the expected configuration. 

That's why I'm comfortable taking some shortcuts in the internals as long as these don't leak.

Reply:
I pushed some cleanup here. I'm not dogmatic when it comes to encapsulation, I was mainly concerned with reading values from the first node in order to pass to the constructors of subsequent nodes. Just seemed we were opening up visibility on these things simply to facilitate that hack. I think my changes make things a little more clear.